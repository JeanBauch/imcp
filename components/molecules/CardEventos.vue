<template>
	<div class="flex flex-col p-4 bg-white rounded-2xl gap-4 border border-primary-beige-500 shadow-sm min-w-[46%] group">
		<div class="flex items-center box-border gap-4">
			<DateOnCardEventos :bg-color="bgColor" :date="content.date" />
			<h4 :class="textColor" class="text-lg sm:text-xl text-left">
				{{ content.title.toUpperCase() }}
			</h4>
		</div>
		<div class="flex items-center">
			<div>
				<img :src="`icons/small-icons-colors/${color}/location.svg`" alt="Icone de marcação de Localização">
			</div>
			<p :class="textColor" class="ml-4 text-sm sm:text-base">
				{{ content.location.address }}
			</p>
		</div>
		<div class="flex">
			<div>
				<img :src="`icons/small-icons-colors/${color}/clock.svg`" alt="Icone de Relógio">
			</div>
			<p :class="textColor" class="ml-4 text-sm sm:text-base">
				{{ content.date.hour }}
			</p>
		</div>
	</div>
</template>

<script lang="ts">
import Vue from 'vue';
import DateOnCardEventos from '../atoms/DateOnCardEventos.vue';

export default Vue.extend({
	data () {
		return {
			bgColor: `bg-${this.color === 'blue' ? 'secondary' : 'primary'}-${this.color}-500`,
			textColor: `text-${this.color === 'blue' ? 'secondary' : 'primary'}-${this.color}-500`,
			content: {
				date: {
					day: '24',
					month: 'abril',
					hour: '10h',
				},
				title: '33 ANOS DO CORAL DAS MENINAS CANTORAS DE PETRÓPOLIS',
				location: {
					address: 'Rua Santos Dumont, 355 | Petrópolis - RJ - Brasil',
				},
			},
		};
	},
	props: {
		color: {
			required: true,
			type: String,
			default: 'yellow',
			validator: (value: string) => [
				'bordo',
				'blue',
			].includes(value.toLowerCase()),
		},
	},
	components: { DateOnCardEventos },
});
</script>
