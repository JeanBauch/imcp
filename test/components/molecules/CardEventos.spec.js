import { shallowMount } from '@vue/test-utils';

import CardEventos from '~/components/molecules/CardEventos.vue';

describe('Components:organisms:CardEvents', () => {
	const factory = (values = {}) => {
		return shallowMount(CardEventos, {
			propsData: {
				...values,
			},
		});
	};
	describe('Snapshots', () => {
		it('should render component', () => {
			const wrapper = factory({ color: 'bordo' });

			expect(wrapper.html()).toMatchSnapshot();
		});

		it('should render correctly data as with props datas', () => {
			const wrapper = shallowMount(CardEventos, {
				propsData: {
					color: 'bordo',
				},
			});

			expect(wrapper.find('.text-primary-bordo-500').exists()).toBeTruthy();
		});

		it('should render correctly data with bordo or blue', async () => {
			const wrapper = factory({ color: 'blue' });

			await wrapper.setData({ bgColor: 'bg-secondary-blue-500' });

			expect(wrapper.vm.bgColor).toBe('bg-secondary-blue-500');
		});
	});
});
