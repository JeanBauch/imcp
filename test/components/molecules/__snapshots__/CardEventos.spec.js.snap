// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components:organisms:CardEvents Snapshots should render component 1`] = `
"<article class=\\"flex flex-col p-4 bg-white rounded-2xl gap-4 border border-primary-beige-500 shadow-sm min-w-[46%] group\\">
  <div class=\\"flex items-center box-border gap-6\\">
    <dateoncardeventos-stub date=\\"[object Object]\\" bgcolor=\\"bg-primary-bordo-500\\"></dateoncardeventos-stub>
    <h4 class=\\"text-lg sm:text-xl text-left text-primary-bordo-500\\">
      33 ANOS DO CORAL DAS MENINAS CANTORAS DE PETRÓPOLIS
    </h4>
  </div>
  <div class=\\"flex items-center\\">
    <div><img src=\\"icons/small-icons-colors/bordo/location.svg\\" alt=\\"Icone de marcação de Localização\\"></div>
    <p class=\\"ml-4 text-sm sm:text-base text-primary-bordo-500\\">
      Rua Santos Dumont, 355 | Petrópolis - RJ - Brasil
    </p>
  </div>
  <div class=\\"flex\\">
    <div><img src=\\"icons/small-icons-colors/bordo/clock.svg\\" alt=\\"Icone de Relógio\\"></div>
    <p class=\\"ml-4 text-sm sm:text-base text-primary-bordo-500\\">
      10h
    </p>
  </div>
</article>"
`;
